compaction/compaction_strategy.cc

```diff
@@ -87,17 +87,63 @@ std::optional<sstring> compaction_strategy_impl::get_value(const std::map<sstrin
     return it->second;
 }
 
-compaction_strategy_impl::compaction_strategy_impl(const std::map<sstring, sstring>& options) {
-    using namespace cql3::statements;
+void compaction_strategy_impl::validate_options_for_strategy_type(const std::map<sstring, sstring>& options, sstables::compaction_strategy_type type) {
+    switch (type) {
+        case compaction_strategy_type::size_tiered:
+            size_tiered_compaction_strategy::validate_options(options);
+            break;
+        case compaction_strategy_type::leveled:
+            leveled_compaction_strategy::validate_options(options);
+            break;
+        case compaction_strategy_type::time_window:
+            time_window_compaction_strategy::validate_options(options);
+            break;
+        default:
+            break;
+    }
```

Comment:
every compaction strategy inherits the basic options, i.e. compaction_strategy_impl::validate_options()

so perhaps we can call it unconditionally once here? therefore, we reduce boilerplate for a strategy-specific validation, which will not have to worry about calling compaction_strategy_impl::validate_options().